src/OpenSpaceToolkit/Astrodynamics/NumericalSolver.cpp

```diff
@@ -171,99 +175,137 @@ Real                            NumericalSolver::getAbsoluteTolerance       ( )
 
 }
 
-NumericalSolver::StateVector    NumericalSolver::integrateStateForDuration  (   const   StateVector&                anInitialStateVector,
-                                                                                const   Duration&                   anIntegrationDuration,
-                                                                                const   NumericalSolver::SystemOfEquationsWrapper& aSystemOfEquations           ) const
+Array<NumericalSolver::StateVector> NumericalSolver::integrateStatesAtSortedInstants (  const   StateVector&        anInitialStateVector,
+                                                                                        const   Instant&            aStartInstant,
+                                                                                        const   Array<Instant>&     anInstantArray,
+                                                                                        const   NumericalSolver::SystemOfEquationsWrapper& aSystemOfEquations   )
 {
 
     NumericalSolver::StateVector aStateVector = anInitialStateVector ;
+    states_.clear() ;
```

Comment:
What's the value in using member variables for `states` and `times`?
Why not instantiate an object that's scope for the method and then make the method a const?